lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -267,9 +267,26 @@ static LogicalResult verifyCircuitOp(CircuitOp circuit) {
   // that defines it or, preferentially, the first external module
   // that defines it and has no parameters.
   llvm::DenseMap<Attribute, FExtModuleOp> defnameMap;
+  InnerRefList instanceSyms;
+  InnerRefList leafInnerSyms;
+  SmallVector<NonLocalAnchor> nlaList;
 
   // Verify external modules.
   for (auto &op : *circuit.getBody()) {
+    // Record all the NLAs.
+    if (auto nla = dyn_cast<NonLocalAnchor>(op))
+      nlaList.push_back(nla);
+    else if (auto mod = dyn_cast<FModuleOp>(op)) {
+      auto modName = mod.getNameAttr();
+      mod.walk([&](Operation *op) {
+        if (isa<InstanceOp>(op))
+          instanceSyms.insert(op, modName);
+        else
+          leafInnerSyms.insert(op, modName);
+      });
+    }
+    if (auto mod = dyn_cast<FModuleLike>(op))
+      leafInnerSyms.insert(mod);
```

Comment:
I feel like it might be better to move the `ExtModule` logic into its own function, so this part can be:
```c++
  for (auto &op : *circuit.getBody()) {
    // Record all the NLAs.
    if (auto nla = dyn_cast<NonLocalAnchor>(op))
      nlaList.push_back(nla);
    else if (auto mod = dyn_cast<FModuleOp>(op)) {
      leafInnerSyms.insert(mod);  // <- this line was moved here from below.
      auto modName = mod.getNameAttr();
      mod.walk([&](Operation *op) {
        instanceSyms.insert(op, modName);
      });
    } else if (auto extModule = dyn_cast<FExtModuleOp>(op)) {
      verifyExtModuleDefname(defnameMap, extModule);
    }
  }
```

Reply:
Good idea, moved the check out.